docs_user/modules/openstack-swift_adoption.adoc

```diff
@@ -0,0 +1,150 @@
+[id="adopting-the-object-storage-service_{context}"]
+
+//:context: adopting-object-storage-service
+
+= Adopting the Object Storage service
+
+== Limitations
+
+* The described process does not migrate data from existing nodes yet. Data is
+  still stored on existing nodes, but is accessed through the Swift proxy
+  instance running on the OpenShift control plane.
+
+== Prerequisites
+
+* Previous adoption steps completed.
+* The Swift storage backend services must NOT be stopped.
+* Storage network has been properly configured on the OpenShift cluster.
+
+== Variables
+
+No new environmental variables need to be defined, though you use the
+`CONTROLLER1_SSH` that was defined in a previous step for the pre-checks.
+
+== Pre-checks
+
+== Copy over swift.conf file
+
+* Create the `swift-conf` secret, containing the Swift hash path suffix and prefix:
++
+[source,yaml]
+----
+oc apply -f - <<EOF
+apiVersion: v1
+kind: Secret
+metadata:
+  name: swift-conf
+  namespace: openstack
+type: Opaque
+data:
+  swift.conf: $($CONTROLLER1_SSH cat /var/lib/config-data/puppet-generated/swift/etc/swift/swift.conf | base64 -w0)
+EOF
+----
+
+== Copy existing Swift ring files
+
+* Create the `swift-ring-files` configmap, containing the Swift ring files:
++
+[source,yaml]
+----
+oc apply -f - <<EOF
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: swift-ring-files
+binaryData:
+  swiftrings.tar.gz: $($CONTROLLER1_SSH "cd /var/lib/config-data/puppet-generated/swift/etc/swift && tar cz *.builder *.ring.gz backups/ | base64 -w0")
+EOF
+----
+
+== Procedure - Swift adoption
+
+* Patch OpenStackControlPlane to deploy Swift:
++
+[source,yaml]
+----
+oc patch openstackcontrolplane openstack --type=merge --patch '
+spec:
+  swift:
+    enabled: true
+    template:
+      secret: osp-secret
+      swiftConfSecret: swift-conf
+      memcachedInstance: memcached
+      swiftRing:
+        ringReplicas: 1
+      swiftStorage:
+        replicas: 0
+        networkAttachments:
+        - storage
+        storageClass: local-storage
+        storageRequest: 10Gi
+      swiftProxy:
+        secret: osp-secret
+        replicas: 1
+        passwordSelectors:
+          service: SwiftPassword
+        serviceUser: swift
+        override:
+          service:
+            internal:
+              metadata:
+                annotations:
+                  metallb.universe.tf/address-pool: internalapi
+                  metallb.universe.tf/allow-shared-ip: internalapi
+                  metallb.universe.tf/loadBalancerIPs: 172.17.0.80
+              spec:
+                type: LoadBalancer
+        networkAttachments:
+        - storage
+'
+----
+
+== Post-checks
+
+=== Inspect the resulting swift service pods
+
+[,bash]
+----
+oc get pods -l app.kubernetes.io/name=SwiftProxy
```

Comment:
This needs to be changed once https://github.com/openstack-k8s-operators/swift-operator/pull/126 is merged and the openstack-operator bumped the swift-operator version. Should be "oc get pods -l component=swift-proxy" then

Reply:
Done